src/cj.c

```diff
@@ -4,20 +4,26 @@ SEXP cj(SEXP base_list) {
   int JJ = LENGTH(base_list), nprotect = 0;
   SEXP out = PROTECT(allocVector(VECSXP, JJ)); nprotect++;
   int NN = 1;
+  // already confirmed to be less than .Machine$integer.max in R
```

Comment:
so we compute it twice? maybe handle this here only

Reply:
Was looking at that, but then we'd have to either (a) move the sorting step to C or (b) do the sorting only to `stop` right afterwards. `NN` should be almost instant to compute of course...

Reply:
we can also pass that value from R to C as extra attribute, but not sure if it is worth as it will be instant computation anyway.

Reply:
exactly... complicating the interface for micro gain, unless `...` is hundreds of columns, in which case almost surely the `integer.max` limit will be hit anyway unless most of `...` has length 1... seems unlikely.

Reply:
For porting the sort to C... fine I guess, but there's the funky bit about generating `dups` there as well, would have to return `dups` as an attribute on the C output, sounds clunky & again probably unnecessary...